config.js

```diff
@@ -0,0 +1,132 @@
+import { sparqlEscapeUri } from 'mu';
+
+export const UPDATE_BATCH_SIZE = parseInt(process.env.UPDATE_BATCH_SIZE) || 10;
+export const SELECT_BATCH_SIZE = parseInt(process.env.SELECT_BATCH_SIZE) || 1000;
+export const COLLABORATOR_GRAPH_PREFIX = process.env.COLLABORATOR_GRAPH_PREFIX || 'http://mu.semte.ch/graphs/organizations/';
+
+export const PREFIXES = `
+    PREFIX mu: ${sparqlEscapeUri('http://mu.semte.ch/vocabularies/core/')}
+    PREFIX ext: ${sparqlEscapeUri('http://mu.semte.ch/vocabularies/ext/')}
+    PREFIX ebucore: ${sparqlEscapeUri('http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#')}
+    PREFIX prov: ${sparqlEscapeUri('http://www.w3.org/ns/prov#')}
+    PREFIX vcard: ${sparqlEscapeUri('http://www.w3.org/2006/vcard/ns#')}
+    PREFIX foaf: ${sparqlEscapeUri('http://xmlns.com/foaf/0.1/')}
+    PREFIX fabio: ${sparqlEscapeUri('http://purl.org/spar/fabio/')}
+    PREFIX session: ${sparqlEscapeUri('http://mu.semte.ch/vocabularies/session/')}
+    PREFIX nmo: ${sparqlEscapeUri('http://www.semanticdesktop.org/ontologies/2007/03/22/nmo#')}
+    PREFIX nie: ${sparqlEscapeUri('http://www.semanticdesktop.org/ontologies/2007/01/19/nie#')}
+    PREFIX nfo: ${sparqlEscapeUri('http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#')}
+    PREFIX dct: ${sparqlEscapeUri('http://purl.org/dc/terms/')}
+    PREFIX dbpedia: ${sparqlEscapeUri('http://purl.org/dc/terms/')}
+    PREFIX skos: ${sparqlEscapeUri('http://www.w3.org/2004/02/skos/core#')}
+    PREFIX adms: ${sparqlEscapeUri('http://www.w3.org/ns/adms#')}
+`;
+
+export const PRESS_RELEASE_PROPERTIES = {
+    'resourcePredicate': 'fabio:PressRelease',
+    'predicates': [
+        'http://www.semanticdesktop.org/ontologies/2007/01/19/nie#title',
+        'http://www.semanticdesktop.org/ontologies/2007/01/19/nie#htmlContent',
+        'http://purl.org/dc/terms/abstract',
+        'http://www.semanticdesktop.org/ontologies/2007/01/19/nie#keyword',
+        'http://purl.org/dc/terms/created',
+        'http://purl.org/dc/terms/modified',
+        'http://purl.org/dc/terms/creator',
+    ],
+};
+
+export const LINKED_RESOURCES = [
+    {
+        // Attachments
+        'parentPredicate': 'fabio:PressRelease',
+        'relationPredicate': 'nie:hasPart',
+        'resourcePredicate': 'nfo:FileDataObject',
+        'predicates': [
+            'http://purl.org/dc/terms/created',
+            'http://purl.org/dc/terms/modified',
+            'http://purl.org/dc/terms/format',
+            'http://mu.semte.ch/vocabularies/core/uuid',
+            'http://dbpedia.org/ontology/fileExtension',
+            'http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#fileName',
+            'http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#fileSize',
+            'http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#fileCreated',
+        ],
+        'relations': [
```

Comment:
Je maakt het hier complex met die geneste relaties. Je kan dit generieke opvatten als je je lostrekt van het onderscheid properties/relaties. In een triplestore zijn alles triples.

Wat we basically willen is voor een aantal resource types, die via een link gerelateerd zijn aan een persbericht, een aantal predicates/objects kopieren. Je kan dat configureren in een flat-list met een item voor ieder resource type waarbij je bij ieder type specifieert:
- wat het path is naar het persbericht. Indien dit path meer dan 1 segment is (de 'nested' relations) kan je hier SPARQL property paths voor gebruiken
- welke properties je voor de resource wil kopieren

Als je dat voor elke resource doet, gaan de triples die de resources linken aan elkaar automatisch meegekopieerd worden.

De code om de temp-graph op te stellen is dan niet meer dan 1 for-loop over die config array en voor ieder resource type de insert data query genereren.